modules/ROOT/pages/clauses/match.adoc

```diff
@@ -490,3 +490,135 @@ The above query uses the xref:functions/aggregating.adoc#functions-collect[`coll
 
 For more information about how Cypher queries work, see xref:clauses/clause-composition.adoc[].
 
+[role=label--new-5.26]
+[[dynamic-match]]
+== MATCH nodes and relationships using dynamic node labels and relationship types
+
+Node labels and relationship types can be referenced dynamically in parameters and variables when matching nodes and relationships.
+
+.Syntax for matching node labels dynamically
+[source, syntax]
+----
+MATCH (n:$(<expr>))
+MATCH (n:$any(<expr>))
+MATCH (n:$all(<expr>))
+----
+
+.Syntax for matching relationship types dynamically
+[source, syntax]
+----
+MATCH ()-[r:$(<expr>))]->()
+MATCH ()-[r:$any(<expr>)]->()
+MATCH ()-[r:$all(<expr>))]->()
+----
+
+The expression must evaluate to a `STRING NOT NULL | LIST<STRING NOT NULL> NOT NULL` value.
+
+.Match labels dynamically using the `all()` function
+[source, cypher]
+----
+WITH "Movie" AS label
+MATCH (movie:$all(label))
+RETURN movie AS movieNodes
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| movieNodes
+
+| (:Movie {title: "Wall Street"})
+| (:Movie {title: "The American President"}) 
+
+1+d|Rows: 2
+|===
+
+[NOTE]
+The xref:functions/predicate.adoc#functions-all[`all()`] function matches nodes that have all the specified labels.
+
+If passing a `LIST<STRING>` to an `all()` function evaluating a relationship pattern using dynamic relationship types, the list cannot contain more than one element.
```

Comment:
It can, it just won't match on anything (emitting a warning is not the same as disallowed) :)